kommunicate/src/main/java/io/kommunicate/Kommunicate.java

```diff
@@ -158,18 +160,155 @@ public void onFailure(RegistrationResponse registrationResponse, Exception excep
         };
     }
 
-    public static void login(Context context, KMUser kmUser, KMLoginHandler handler, ResultReceiver prechatReceiver) {
+    public static void login(final Context context, final KMUser kmUser, final KMLoginHandler handler, ResultReceiver prechatReceiver) {
         if (kmUser != null) {
             kmUser.setHideActionMessages(true);
             kmUser.setSkipDeletedGroups(true);
         }
+
         new KmUserLoginTask(kmUser, false, handler, context, prechatReceiver).execute();
     }
 
+    public static void launchLeadCollection(final Context context, List<KmPrechatInputModel> inputModelList, final KmPrechatCallback<KMUser> callback) throws KmException {
+        if (!(context instanceof Activity)) {
+            throw new KmException("This method needs Activity context");
+        }
+
+        ResultReceiver resultReceiver = new ResultReceiver(null) {
+            @Override
+            protected void onReceiveResult(int resultCode, Bundle resultData) {
+                if (KmConstants.PRECHAT_RESULT_CODE == resultCode) {
+                    KMUser user = (KMUser) GsonUtils.getObjectFromJson(resultData.getString(KmConstants.KM_USER_DATA), KMUser.class);
+                    if (callback != null) {
+                        callback.onReceive(user, context, (ResultReceiver) resultData.getParcelable(KmConstants.FINISH_ACTIVITY_RECEIVER));
+                    }
+                }
+            }
+        };
+
+        try {
+            Intent intent = new Intent(context, KmUtils.getClassFromName(KmConstants.PRECHAT_ACTIVITY_NAME));
+            intent.putExtra(KmPrechatInputModel.KM_PRECHAT_MODEL_LIST, GsonUtils.getJsonFromObject(inputModelList, List.class));
+            intent.putExtra(KmConstants.PRECHAT_RESULT_RECEIVER, resultReceiver);
+            context.startActivity(intent);
+        } catch (ClassNotFoundException e) {
+            throw new KmException(e.getMessage());
+        }
+    }
+
     public static void loginAsVisitor(Context context, KMLoginHandler handler) {
         login(context, getVisitor(), handler);
     }
 
+    public static void loginAsVisitorV2(final Context context, final KmCallback callback) {
```

Comment:
rather than using "loginAsVisitorV2", we can use "loginWithPrechat"

Reply:
Yeah, this is for our reference not the final method name.Once the code testing & review is will update the name as **launchConversationWithPreChat()**